diff --git a/init/property_service.cpp b/init/property_service.cpp
index 2d67bf5..56509b5 100644
--- a/init/property_service.cpp
+++ b/init/property_service.cpp
@@ -189,8 +189,8 @@ static uint32_t PropertySet(const std::string& name, const std::string& value, s
     if (pi != nullptr) {
         // ro.* properties are actually "write-once".
         if (StartsWith(name, "ro.")) {
-            *error = "Read-only property was already set";
-            return PROP_ERROR_READ_ONLY_PROPERTY;
+            //*error = "Read-only property was already set";
+            //return PROP_ERROR_READ_ONLY_PROPERTY;
         }
 
         __system_property_update(pi, value.c_str(), valuelen);
